### PR TITLE
Add announcement to DateInput when toggling calendar dialog

### PR DESCRIPTION
**Pull request by jeferson-sb**
#### What does this PR do?

Add an accessibility announcement to DateInput so when users toggle the calendar they get alerted and know whether or not it is opened.

#### Where should the reviewer start?
`src/js/components/DateInput/DateInput.js`

#### What testing has been done on this PR?
`yarn test`

#### How should this be manually tested?
through storybook with a Screen reader or VoiceOver enabled.

#### What are the relevant issues?
See #5437 
Also related to #4068 

#### Do the grommet docs need to be updated?
No

#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/DateInput/DateInput.js [hunk 1]
```diff
@@ -92,6 +95,16 @@ const DateInput = forwardRef(
     // when format and not inline, whether to show the Calendar in a Drop
     const [open, setOpen] = useState();
 
+    const openCalendar = useCallback(() => {
+      setOpen(true);
+      announce('Enter calendar dialog');
```
Comment (MikeKingdom) [inline src/js/components/DateInput/DateInput.js:100]: This text needs to be internationalized. See #5468 as an example

Comment (jeferson-sb) [reply src/js/components/DateInput/DateInput.js]: Good call. Thanks!

File: src/js/components/DateInput/DateInput.js [hunk 2]
```diff
@@ -40,11 +43,14 @@ const DateInput = forwardRef(
...
       onFocus,
       value: valueArg,
+      messages,
       ...rest
     },
     refArg,
   ) => {
     const theme = useContext(ThemeContext) || defaultProps.theme;
+    const announce = useContext(AnnounceContext);
+    const { format: formatMessage } = useContext(MessageContext);
```
Comment (britt6612) [inline src/js/components/DateInput/DateInput.js:53]: To stay constant can we do `const { format } = useContext(MessageContext);`

File: src/js/components/DateInput/DateInput.js [hunk 3]
```diff
@@ -92,6 +98,16 @@ const DateInput = forwardRef(
     // when format and not inline, whether to show the Calendar in a Drop
     const [open, setOpen] = useState();
 
+    const openCalendar = useCallback(() => {
+      setOpen(true);
+      announce(formatMessage({ id: 'dateInput.enterCalendar', messages }));
+    }, [announce, formatMessage, messages]);
+
+    const exitCalendar = useCallback(() => {
```
Comment (taysea) [inline src/js/components/DateInput/DateInput.js:106]: naming suggestion: `closeCalendar` to be consistent with state variable of `setOpen` (open/close)

File: src/js/components/DateInput/DateInput.js (hunk 2 above)
Comment (jeferson-sb) [reply src/js/components/DateInput/DateInput.js:53]: @britt6612 There's an existing `format` prop on this component

Comment (britt6612) [reply src/js/components/DateInput/DateInput.js:53]: @jeferson-sb my mistake just saw that thank you for clarifying! 👍

Comment (britt6612) [reply src/js/components/DateInput/DateInput.js:53]: did you run `yarn test -update`?

File: src/js/languages/default.json [hunk 4]
```diff
@@ -48,5 +48,9 @@
     "playButton": "play",
     "volumeDown": "volume down",
     "volumeUp": "volume up"
+  },
+  "dateInput": {
+    "enterCalendar": "Enter calendar dialog",
```
Comment (taysea) [inline src/js/languages/default.json:53]: Should we do something slightly more descriptive like the one above for textinput. Maybe "Calendar is open, press tab to enter the calendar."

I feel like that might create better context for someone using a screen reader.

Comment (jeferson-sb) [reply src/js/languages/default.json:53]: @taysea yeah, I think that makes more sense.

**Comment by britt6612** [comment]
@jeferson-sb Looks good! 
I updated snapshots so @taysea you can take one final look 👍

**Review by taysea — APPROVED** [review]
LGTM. Just need to update docs on the grommet-site repo since we've changed how docs are created. : )

File: src/js/components/Button/__tests__/__snapshots__/Button-kind-test.js.snap [hunk 5]
```diff
@@ -872,7 +872,7 @@ exports[`Button kind badge should render custom element 1`] = `
           viewBox="0 0 24 24"
         >
           <path
-            d="M12 22V2M2 12h20"
+            d="M12,22 L12,2 M2,12 L22,12"
```
Comment (ericsoderberghp) [inline src/js/components/Button/__tests__/__snapshots__/Button-kind-test.js.snap:875]: Why is there a snapshot change in this SVG?

Comment (jeferson-sb) [reply src/js/components/Button/__tests__/__snapshots__/Button-kind-test.js.snap:875]: @ericsoderberghp I'm not sure, it came when I ran `yarn test -update`

Comment (halocline) [reply src/js/components/Button/__tests__/__snapshots__/Button-kind-test.js.snap:875]: @jeferson-sb I suspect you need to make sure you are referencing the latest Grommet dependencies, specifically grommet-icons. Can you make sure you have run `yarn cache clean && yarn` recently? Then run the `yarn test --update`.

Suspicion is due to how the paths are changing and are no longer SVG optimized.

Comment (jeferson-sb) [reply src/js/components/Button/__tests__/__snapshots__/Button-kind-test.js.snap]: @halocline cool, that works! thank you

**Comment by britt6612** [comment]
@jeferson-sb LGTM thanks for cleaning this up

**Review by ericsoderberghp — APPROVED** [review]

**Review by MikeKingdom — APPROVED** [review]
Looks great

**Pull request merged**